### PR TITLE
Refactors xeno eggs, and makes glowing resin glow

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: Alien glowing resin now glows.
/:cl:

Christ, all I wanted was to make glowing resin glow blue, like it says
it's supposed to.

Instead, I stumble onto the horrors of alien egg code.
First fix the spawn, then fix the icon states, then fix var edited burst
eggs, then make fully grown eggs spawned, then fix some weird use of
locate, rather than a var, then make the typecache spawn on Initialize.

Should be no behaviour differences. Apart from the glowing resin.

### CODE REVIEW

File: code/game/objects/structures/aliens.dm [hunk 1]
```diff
@@ -171,12 +176,14 @@
 	name = "glowing resin"
 	desc = "Blue bioluminescence shines from beneath the surface."
 	icon_state = "weednode"
-	light_range = 1
+	light_color = LIGHT_COLOR_BLUE
```
Comment (ChangelingRain) [inline code/game/objects/structures/aliens.dm]: I think LIGHT_CYAN might be better here; please get pictures

**Comment by AnturK** [comment]
Pictures.

File: code/game/objects/structures/aliens.dm [hunk 2]
```diff
@@ -200,39 +207,53 @@
...
+		add_to_proximity_list(src, 1)
 
 /obj/structure/alien/egg/Destroy()
 	remove_from_proximity_list(src, 1)
-	return ..()
+	. = ..()
+
+/obj/structure/alien/egg/update_icon()
+	..()
+	if(status == GROWING)
```
Comment (RemieRichards) [inline code/game/objects/structures/aliens.dm]: switch

File: code/game/objects/structures/aliens.dm [hunk 3]
```diff
@@ -200,39 +207,53 @@
...
+	var/obj/item/clothing/mask/facehugger/child
 
 
-/obj/structure/alien/egg/New()
-	new /obj/item/clothing/mask/facehugger(src)
+/obj/structure/alien/egg/Initialize(mapload)
 	..()
-	addtimer(CALLBACK(src, .proc/Grow), rand(MIN_GROWTH_TIME, MAX_GROWTH_TIME))
+	update_icon()
+	if(status == GROWING || status == GROWN)
```
Comment (RemieRichards) [inline code/game/objects/structures/aliens.dm]: switch

**Comment by lzimann** [comment]
👍 for refactor 
👎 for even more colored lights

File: code/game/objects/structures/aliens.dm [hunk 4]
```diff
@@ -127,15 +127,20 @@
...
-	/turf/open/floor/plating/lava))
+	var/static/list/blacklisted_turfs
 
-/obj/structure/alien/weeds/New()
+/obj/structure/alien/weeds/Initialize()
 	pixel_x = -4
 	pixel_y = -4 //so the sprites line up right in the map editor
 	..()
+
+	if(!blacklisted_turfs)
```
Comment (RemieRichards) [inline code/game/objects/structures/aliens.dm]: there's no improvement to moving this here, put it back.

Comment (coiax) [inline code/game/objects/structures/aliens.dm]: But I saw other things do it, something about... lights on New, rather than putting them in Initialize?

Comment (RemieRichards) [inline code/game/objects/structures/aliens.dm]: well this isn't lights, it's a static list (typecache) that's generated once

**Comment by FantasticFwoosh** [comment]
For lore, why not have inactive eggs luminate very translucent thin smoke?

File: code/game/objects/structures/aliens.dm (hunk 3 above)
Comment (coiax) [inline code/game/objects/structures/aliens.dm]: How do you do multi test switches?

Comment (RemieRichards) [inline code/game/objects/structures/aliens.dm]: ```if(A, B)```

**Comment by WJohn** [comment]
Coiax showed this pic in coderbus between cyan on the left and blue on the right.

Personally I prefer the blue on the right.

**Comment by GunHog** [comment]
I also prefer the blue!

File: code/game/objects/structures/aliens.dm [hunk 5]
```diff
@@ -310,7 +331,15 @@
...
+/obj/structure/alien/egg/grown
+	status = GROWN
+	icon_state = "egg"
+
+/obj/structure/alien/egg/burst
+	status = BURST
+	icon_state = "egg_hatched"
 
 #undef BURST
 #undef BURSTING
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/objects/structures/aliens.dm]: I think you can remove BURSTING

Comment (coiax) [inline code/game/objects/structures/aliens.dm]: Nice catch.

**Comment by Sligneris** [comment]
Meanwhile I prefer cyan.

**Pull request merged**